lib/trino-filesystem/src/main/java/io/trino/filesystem/TrinoFileSystem.java

```diff
@@ -63,6 +64,18 @@ public interface TrinoFileSystem
      */
     TrinoInputFile newInputFile(Location location);
 
+    /**
+     * Creates an encrypted TrinoInputFile which can be used to read the encrypted file data.
+     * The file location path cannot be empty, and must not end with a slash or whitespace.
+     *
+     * @throws IllegalArgumentException if location is not valid for this file system
+     * @throws UnsupportedOperationException if server side encryption is not supported
+     */
+    default TrinoInputFile newEncryptedInputFile(Location location, EncryptionKey key)
```

Comment:
The current PR is missing a broader context on where is this building block going to be used.

This could help a potential reviewer to connect the dots easier on why do we need to extend the `TrinoFileSystem` interface.

Reply:
it will be used in the spooling. I've discussed API with @electrum already